lib/wit/deps/fastly/compute.wit

```diff
@@ -0,0 +1,1076 @@
+
+  type handle = u32;
+
+  endpoint-get: func(name: string) -> result<handle, error>;
+
+  write: func(h: handle, msg: string) -> result<u64, error>;
+}
+
+/*
+ * Fastly HTTP Req
+ */
+interface http-req {
+
+  use types.{error};
+  use http-types.{
+    body-handle, request-handle, http-version, response, pending-request-handle,
+    content-encodings, framing-headers-mode, backend-config-options,
+    dynamic-backend-config
+  };
+
+  flags cache-override-tag {
+    /// Do not cache the response to this request, regardless of the origin response's headers.
+    pass,
+    ttl,
+    stale-while-revalidate,
+    pci,
+  }
+
+  enum client-cert-verify-result {
+    ok,
+    bad-certificate,
+    certificate-revoked,
+    certificate-expired,
+    unknown-ca,
+    certificate-missing,
+    certificate-unknown,
+  }
+
+  enum send-error-detail-tag {
+      uninitialized,
+      ok,
+      dns-timeout,
+      dns-error,
+      destination-not-found,
+      destination-unavailable,
+      destination-ip-unroutable,
+      connection-refused,
+      connection-terminated,
+      connection-timeout,
+      connection-limit-reached,
+      tls-certificate-error,
+      tls-configuration-error,
+      http-incomplete-response,
+      http-response-header-section-too-large,
+      http-response-body-too-large,
+      http-response-timeout,
+      http-response-status-invalid,
+      http-upgrade-failed,
+      http-protocol-error,
+      http-request-cache-key-invalid,
+      http-request-uri-invalid,
+      internal-error,
+      tls-alert-received,
+      tls-protocol-error,
+  }
+
+  flags send-error-detail-mask {
+      reserved,
+      dns-error-rcode,
+      dns-error-info-code,
+      tls-alert-id,
+  }
+
+  record send-error-detail {
+      tag: send-error-detail-tag,
+      mask: send-error-detail-mask,
+      dns-error-rcode: u16,
+      dns-error-info-code: u16,
+      tls-alert-id: u8,
+  }
+
+  cache-override-set: func(
+    h: request-handle,
+    tag: cache-override-tag,
+    ttl: u32,
+    stale-while-revalidate: u32,
+  ) -> result<_, error>;
+
+  cache-override-v2-set: func(
+    h: request-handle,
+    tag: cache-override-tag,
+    ttl: u32,
+    stale-while-revalidate: u32,
+    sk: option<string>
+  ) -> result<_, error>;
+
+  downstream-client-ip-addr: func() -> result<list<u8>, error>;
+
+  downstream-server-ip-addr: func() -> result<list<u8>, error>;
+
+  downstream-client-h2-fingerprint: func(max-len: u64) -> result<list<u8>, error>;
+
+  downstream-client-request-id: func(max-len: u64) -> result<string, error>;
+
+  downstream-client-oh-fingerprint: func(max-len: u64) -> result<list<u8>, error>;
+
+  downstream-tls-cipher-openssl-name: func(max-len: u64) -> result<string, error>;
+
+  downstream-tls-protocol: func(max-len: u64) -> result<string, error>;
+
+  downstream-tls-client-hello: func(max-len: u64) -> result<list<u8>, error>;
+
+  downstream-tls-raw-client-certificate: func(max-len: u64) -> result<list<u8>, error>;
```

Comment:
Is there any way we could validate that these match the `witx` definitions, or did we generate this file in an automated way that we trust?

I admit that my eyes started glazing over right about here, and the chance that I can do a good, thorough review that this matches the existing spec is right around 0%.

Reply:
They're not going to match it 100%, and I'm not sure there's a good way we can ensure that. I'd be happy to brainstorm a bit on how we might simplify the process of checking it though!

Reply:
I guess, in the end, users of the existing ABI will need to go through the adapter layer regardless, so as long as that adapter is capable of any minor translations required, we should be good.